src/project_name/foo.py

```diff
@@ -3,5 +3,19 @@
 
 
 def foo():
-    """Its here, so the directory gets pushed. You are free to move it"""
+    """
+    Its here, so the directory gets pushed. You are free to move it.
+    
+    An example usage can be shown with a doctest:
+
+    >>> from project_name.foo import foo
+    >>> foo()
+    True
+
+    Doctests are included in the test suite and checked in the CI.
+    
+    Doctests can help users understand the code and adding a few examples may not be too hard.
+    However, it still takes time which is often a limiting resource.
+    Therefore, please use your judgment before documenting each and every function.
```

Comment:
You think, people will read this, when its in the docstring? Maybe add this to the read me. What do you think?

Reply:
Yeah, it's a bit longer side of things and it may not be picked up. Maybe this entire thing is just not so important.

I am going to remove the snippet and the entire pydoc text. I think, having tests in the `README.md` is just enough for the template.